C/Clay/build_tarballs.jl

```diff
@@ -0,0 +1,59 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Clay"
+version = v"0.1.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/nicbarker/clay.git", "6a9b723dcce154347452491559b34da528cd657f")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/clay
+echo '#define CLAY_IMPLEMENTATION' > clay.c
+echo '#include "./clay.h"' >> clay.c
+PLATFORM_NAME=$(uname -s)
+if [ "$PLATFORM_NAME" = "Darwin" ]; then
+    mkdir -p ${prefix}/lib
+    clang -shared ${WORKSPACE}/srcdir/clay/clay.c -fPIC -o ${prefix}/lib/libclay.dylib
+elif [ "$PLATFORM_NAME" = "Linux" ] || [ "$PLATFORM_NAME" = "FreeBSD" ]; then
+    mkdir -p ${prefix}/lib
+    clang -shared ${WORKSPACE}/srcdir/clay/clay.c -fPIC -o ${prefix}/lib/libclay.so
+else
+    mkdir -p ${prefix}/bin
+    clang -shared ${WORKSPACE}/srcdir/clay/clay.c -fPIC -o ${prefix}/bin/libclay.dll
+fi
```

Comment:
We have many [automatic environment variables](https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables) to simplify this:
```suggestion
mkdir -p "${libdir}"
cc -shared -fPIC -o "${libdir}/libclay.${dlext}" clay.c
```